File: backend/seed.py
from datetime import datetime, timedelta
from app.database import engine, Base, SessionLocal
from app.models import User, Task
from app.security import get_password_hash

def seed_database():
    print("[*] Initializing MySQL Database Tables...")
    Base.metadata.create_all(bind=engine)

    db = SessionLocal()
    try:
        # Clear existing data
        db.query(Task).delete()
        db.query(User).delete()
        db.commit()

        print("[*] Creating default test user (rupesh@example.com / password123)...")
        hashed_pwd = get_password_hash("password123")
        test_user = User(
            name="Rupesh",
            email="rupesh@example.com",
            password=hashed_pwd
        )
        db.add(test_user)
        db.commit()
        db.refresh(test_user)

        now = datetime.utcnow()
        in_2_hours = now + timedelta(hours=2)
        in_12_hours = now + timedelta(hours=12)
        in_2_days = now + timedelta(days=2)

        print("[*] Populating sample tasks with priorities and deadlines...")
        sample_tasks = [
            Task(
                user_id=test_user.id,
                title="Complete React Native Mobile Assignment",
                description="Build React Native CLI app, FastAPI backend, MySQL database, and Smart Mix Algorithm sorting.",
                date_time=now,
                deadline=in_2_hours,
                priority="urgent",
                category="Study",
                parsed_tags=["React Native", "FastAPI", "MySQL"],
                is_completed=False
            ),
            Task(
                user_id=test_user.id,
                title="Submit Project Repository & Documentation",
                description="Include clean architecture, API routes, and setup instructions.",
                date_time=now,
                deadline=in_12_hours,
                priority="high",
                category="Work",
                parsed_tags=["Git", "FastAPI"],
                is_completed=False
            ),
            Task(
                user_id=test_user.id,
                title="Morning Gym & Fitness Training",
                description="Cardio + strength workout session.",
                date_time=now,
                deadline=in_2_days,
                priority="medium",
                category="Fitness",
                parsed_tags=["Health"],
                is_completed=False
            ),
            Task(
                user_id=test_user.id,
                title="Grocery & Household Supplies",
                description="Buy fresh fruits, milk, and protein bars.",
                date_time=now,
                priority="low",
                category="Shopping",
                parsed_tags=["Home"],
                is_completed=True,
                completed_at=now
            )
        ]

        db.add_all(sample_tasks)
        db.commit()

        print("[SUCCESS] MySQL Database successfully seeded with 1 test user and 4 sample tasks!")
    except Exception as e:
        print("[ERROR] Seeding failed:", e)
        db.rollback()
    finally:
        db.close()

if __name__ == "__main__":
    seed_database()

File: backend/src/utils/seed.ts
import mongoose from 'mongoose';
import bcrypt from 'bcryptjs';
import dotenv from 'dotenv';
import { User } from '../models/User';
import { Task } from '../models/Task';

dotenv.config();

const seedData = async () => {
  try {
    const connStr = process.env.MONGODB_URI || 'mongodb://127.0.0.1:27017/todoapp';
    await mongoose.connect(connStr);
    console.log('Connected to MongoDB for seeding...');

    // Clear existing data
    await User.deleteMany({});
    await Task.deleteMany({});

    // Create Test User
    const salt = await bcrypt.genSalt(10);
    const hashedPassword = await bcrypt.hash('password123', salt);

    const testUser = await User.create({
      name: 'Rupesh',
      email: 'rupesh@example.com',
      password: hashedPassword,
    });

    console.log(`Created test user: ${testUser.email} / password123`);

    const now = new Date();
    const in2Hours = new Date(now.getTime() + 2 * 60 * 60 * 1000);
    const in12Hours = new Date(now.getTime() + 12 * 60 * 60 * 1000);
    const in2Days = new Date(now.getTime() + 2 * 24 * 60 * 60 * 1000);

    // Create Sample Tasks
    await Task.create([
      {
        user: testUser._id,
        title: 'Complete React Native Mobile Assignment',
        description: 'Build user auth, MongoDB API, TypeScript, and smart mix algorithm sorting.',
        dateTime: now,
        deadline: in2Hours,
        priority: 'urgent',
        category: 'Study',
        tags: ['React Native', 'TypeScript', 'Assignment'],
        isCompleted: false,
      },
      {
        user: testUser._id,
        title: 'Submit Code Repository & Documentation',
        description: 'Ensure clean architecture and comprehensive comments.',
        dateTime: now,
        deadline: in12Hours,
        priority: 'high',
        category: 'Work',
        tags: ['Git', 'Backend'],
        isCompleted: false,
      },
      {
        user: testUser._id,
        title: 'Morning Gym & Cardio Workout',
        description: '30 mins treadmill + weight training session.',
        dateTime: now,
        deadline: in2Days,
        priority: 'medium',
        category: 'Fitness',
        tags: ['Health'],
        isCompleted: false,
      },
      {
        user: testUser._id,
        title: 'Weekly Grocery Shopping',
        description: 'Buy fresh fruits, vegetables, and protein shake mix.',
        dateTime: now,
        priority: 'low',
        category: 'Shopping',
        tags: ['Home'],
        isCompleted: true,
        completedAt: now,
      },
    ]);

    console.log('✅ Seed completed successfully with 4 sample tasks!');
    process.exit(0);
  } catch (error) {
    console.error('Error during seeding:', error);
    process.exit(1);
  }
};

seedData();
